lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -451,11 +452,26 @@ private boolean achievedWithGpg45Profile(
                         ? gpg45ProfileEvaluator.getFirstMatchingProfile(
                                 gpg45Scores, requestedVot.getSupportedGpg45Profiles())
                         : Optional.empty();
+        List<SignedJWT> gpg45Credentials = new ArrayList<>();
+        for (SignedJWT credential : credentials) {
+            if (!VcHelper.isOperationalProfileVc(credential)) {
+                gpg45Credentials.add(credential);
+            }
+        }
 
         // Successful match
         if (matchedGpg45Profile.isPresent()) {
+            // remove weaker operational profile
+            if (configService.enabled(INHERITED_IDENTITY.getName())) {
+                for (SignedJWT credential : credentials) {
+                    if (VcHelper.isOperationalProfileVc(credential)) {
+                        verifiableCredentialService.deleteVcStoreItem(
+                                auditEventUser.getUserId(), HMRC_MIGRATION_CRI);
```

Comment:
This is a bit weird - when checking the VC, we're inspecting the VC and using the presence of the `vot` field, but when deleting we're hardcoding `HMRC_MIGRATION_CRI`.

Perhaps the easiest would be to iterate through `vcStoreItems` and delete the `HMRC_MIGRATION_CRI` entry if there is one?

(More generally I think we need to start being more consistent about whether we're using 'presence of VOT' or 'issuer = HMRC Migration' to identify operational profiles - at the moment they're equivalent, but they may not be in future)